lib/site_prism/dsl.rb

```diff
@@ -206,12 +206,25 @@ def create_helper_method(proposed_method_name, *find_args)
       end
 
       def create_rspec_existence_matchers(element_name)
-        matcher = "has_#{element_name}?"
-        negated_matcher = "has_no_#{element_name}?"
+        matcher = "have_#{element_name}"
+        object_method = "has_#{element_name}?"
+        negated_object_method = "has_no_#{element_name}?"
+
+        if RSpec::Matchers::DYNAMIC_MATCHER_REGEX =~ matcher
```

Comment:
@luke-hill so far as I can tell, [this is the rspec code](https://github.com/rspec/rspec-expectations/blob/eb1787f0e1a5ec2c2650048ee60a45d4c9738d78/lib/rspec/matchers.rb#L954-L982) which dynamically handles "have_\<property\>" matchers but it doesn't appear to define concrete methods somewhat that I can introspect for them, so the best I could come up with was logging when defining methods that match the Regex from the method missing implementation.

Reasoning through this based on the matcher format, I believe this line will always be true and thus we'll be logging this message always which seems of questionable value to me. I think I must be misunderstanding your logging request, can you clarify if this is what you meant or something else please?

Reply:
Ok I think you've missed the initial point I was trying to hint at.

Essentially the use case you're quoting is already flawed, because you are encountering a race condition with the positive use case (As well as the negative use case).

What you are saying is, that in this use case (Both are defined), we should want to fall back to one over the other, which I don't quite agree with.

However, your PR also includes a bugfix, which I'm happy to have. Such that, in a use case where only one use case is defined, we should always use just that one use case (This is fine).

So we don't need to test anything in RSpec land. All I'm wanting to do is add a piece of logic that validates where this "requested" method doesn't work. So in the current PR, I would expect a 1line log message at debug level on line 225 Saying that the object didn't respond to SitePrism and it will fall back to using default RSpec logic.


Bringing this full circle, I still don't like this use case, it should never happen. Why someone is mixing page object classes and non-page object classes inside the same testing context I don't know. Especially then using booleans which have (Admittedly not great), similar names to SitePrism's.

Reply:
Thank you for the clarification @luke-hill . I've updated with a log message in the else block